### PR TITLE
Add github codespaces setup

### PR DESCRIPTION
**Pull request by devtomio**
## What does this PR address?

Adds support for developing nixpacks in [GitHub Codespaces](https://github.com/features/codespaces)

## Before submitting:

- [x] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
- [x] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
- [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
- [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
- [ ] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI passes?

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			},
+
+			// Add the IDs of extensions you want installed when the container is created.
+			"extensions": [
+				"ms-azuretools.vscode-docker",
+                "vadimcn.vscode-lldb",
+				"mutantdino.resourcemonitor",
+				"rust-lang.rust-analyzer",
+				"tamasfe.even-better-toml",
+				"serayuzgur.crates"
```
Comment (Milo123459) [inline .devcontainer/devcontainer.json:33]: personally I don't like using this extension, is there a way to disable it from being installed just for me? I looked around the docs and couldn't find it.

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
+FROM rust:1-bullseye
```
Comment (Milo123459) [inline .devcontainer/Dockerfile:1]: Does this come with rustup?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (devtomio) [reply .devcontainer/devcontainer.json:33]: Which one?

Comment (Milo123459) [reply .devcontainer/devcontainer.json:33]: The crates one

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (devtomio) [reply .devcontainer/Dockerfile:1]: Yep, see [this file](https://github.com/rust-lang/docker-rust/blob/master/1.62.1/bullseye/Dockerfile#L24).

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (devtomio) [reply .devcontainer/devcontainer.json:33]: Alright, will remove it. (it is annoying for me too)

Comment (devtomio) [reply .devcontainer/devcontainer.json]: Removed in https://github.com/devtomio/nixpacks/commit/e328915f7fe49828712b224f06c35d0ace58cd1c

**Review by coffee-cup — APPROVED** [review]
I have never used codespaces but if someone else can confirm it works then feel free to merge

**Comment by Milo123459** [comment]
I can test tomorrow, will merge if works.

**Pull request merged**